groth16-framework/test_data/Groth16VerifierExtensions.sol

```diff
@@ -60,24 +77,28 @@ contract Query2 is Verifier {
     // 2. Parse the plonky2 public inputs from the `data` argument.
     // 3. Calculate sha256 on the inputs to a hash value, and set the top 3 bits of this hash to 0.
     //    Then asset this hash value must be equal to the last Groth16 input (groth16_inputs[2]).
-    // 4. Parse a Query instance from the plonky2 public inputs, and asset it must be equal to the
+    // 4. Parse the query identifier from the plonky2 public inputs.
+    // 5. Parse a Query instance from the plonky2 public inputs, and asset it must be equal to the
     //    expected `query` argument.
-    // 5. Parse and return `L` NFT IDs (uint32) from the plonky2 public inputs.
+    // 5. Parse and return the query result from the plonky2 public inputs.
```

Comment:
Shouldn't we have 2 different methods? One for query NTFs and the other one for query ERC-20? With this interface it is unclear to me how the user could specify which query she wants to run, so that we can enforce we used the appropriate circuit. Wdyt?

Reply:
Uhm, you mean to add the identifier as a parameter of `processQuery` method? Isn't it bad for user experience the need to know these constants? That's why I was thinking about 2 methods, so that users can just call `porcessQueryERC20` or `processQueryNTF` without the need to care about these identifiers. But maybe this method is not the end one we expose to users so it's fine to have this low-level details

Reply:
Exactly,  we dont need to expose these details to the user, just internally at query time, we should know wether it's for a erc or nft query so we should be able to store these identifiers at that time already.

Reply:
but that's a discussion to have with @jaeaster, i'll mention it.

Reply:
Added an `identifier` to the `Query` struct in commit [9c4de2b](https://github.com/Lagrange-Labs/Euclid-database/pull/21/commits/9c4de2be84ca84ddce0d53dcf02d806a95b84da4). It's checked with the parsed one from public inputs in the verifier contract.